sklearn/metrics/classification.py

```diff
@@ -1993,3 +1993,64 @@ def brier_score_loss(y_true, y_prob, sample_weight=None, pos_label=None):
     y_true = np.array(y_true == pos_label, int)
     y_true = _check_binary_probabilistic_predictions(y_true, y_prob)
     return np.average((y_true - y_prob) ** 2, weights=sample_weight)
+
+
+def calibration_loss(y_true, y_prob, bin_size=2.0):
+
+    """Compute Calibration score by bins.
+    The calibration loss is defined as the measure to access the quality of
+    learning methods and learned models. A calibration measure based on
+    overlaping binning is CAL (Caruana and Niculescu-Mizil, 2004).
+
+    Parameters
+    ----------
+    y_true : array, shape (n_samples,)
+        True targets.
+
+    y_prob : array, shape (n_samples,)
+        Probabilities of the positive class.
+
+    bin_size : int
+        Size of the bin (samples) analysed in one iteration
+
+    Returns
+    -------
+    score : float
+        Calibration loss
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.metrics import calibration_loss
+    >>> y_true = np.array([0, 1, 1, 0])
+    >>> y_true_categorical = np.array(["spam", "ham", "ham", "spam"])
+    >>> y_prob = np.array([0.1, 0.9, 0.8, 0.3])
+    >>> calibration_loss(y_true, y_prob, bin_size=1)
+    0.175
+    >>> calibration_loss(y_true, y_prob, bin_size=2)
+    0.5333333333333333
+    """
+
+    pos_loss = 0.0
+    neg_loss = 0.0
+
+    for bin_start in range(0, len(y_true)-bin_size + 1):
+
+        bin_end = bin_start + bin_size
+        actual_per_pos_class = (y_true[bin_start:bin_end]
+                                .sum()) / float(bin_size)
```

Comment:
Float should not be needed

Reply:
Np.mean should be preferred anyway

Reply:
For the actual_per_neg_class np.mean couldn't be used, so to maintain consistency, I didn't use it in here as well !

Reply:
Then at least remove float